src/runtime/composables.ts

```diff
@@ -19,114 +19,128 @@ export type { LocaleObject } from 'vue-i18n-routing'
 import type { Locale, LocaleMessages, DefineLocaleMessage, I18nOptions } from 'vue-i18n'
 
```

Comment:
(Style preference if we want to reduce code repetition?)

```suggestion
type RouteOptions<T extends Function, P = 'route' | 'router' | 'i18n'> = Pick<NonNullable<Parameters<T>[0]>, P>
```